modules/ding_place2book/ding_place2book.fields.inc

```diff
@@ -1 +1,298 @@
+      }
+
+      $element['place2book']['capacity'] = [
+        '#type' => 'textfield',
+        '#title' => t('Event capacity'),
+        '#size' => 5,
+        '#maxlength' => 5,
+        '#default_value' => isset($event->capacity) ? $event->capacity : 0,
+        '#description' => t('Capacity of event. 0 - unlimited.'),
+        '#element_validate' => ['element_validate_integer'],
+      ];
+
+      $element['place2book']['passive'] = [
+        '#type' => 'checkbox',
+        '#title' => t('Passive event'),
+        '#default_value' => isset($event->mute) ? $event->mute : 0,
+        '#description' => t('If checked, event will be marked as passive with no ticket sale.'),
+      ];
+
+      $element['place2book']['kultunaut_export'] = [
+        '#type' => 'checkbox',
+        '#title' => t('Kultunaut export'),
+        '#default_value' => isset($event->kultunaut_export) ? $event->kultunaut_export : FALSE,
+        '#description' => t('If checked, Place2book will forward this ding-event to Kultunaut.'),
+      ];
+
+      $options = variable_get('p2b_kultunaut_age_group', []);
+      $element['place2book']['kultunaut_age_group'] = [
+        '#type' => 'select',
+        '#title' => t('Kultunaut age group'),
+        '#options' => $options,
+        '#default_value' => isset($event->kultunaut_age_group) ? $event->kultunaut_age_group : '',
+        '#states' => [
+          'visible' => [
+            ':input[name="field_place2book[und][0][place2book][kultunaut_export]"]' => ['checked' => TRUE],
+          ],
+        ],
+        '#empty_option' => t('- None -'),
+      ];
+
+      if (isset($event->kultunaut_export_category)) {
+        $element['place2book']['kultunaut_export_category'] = [
+          '#markup' => "<p><strong>Kultunaut export category</strong> <span>{$event->kultunaut_export_category}</span></p>",
+        ];
+      }
+
+      $element['ids']['event_id'] = [
+        '#type' => 'value',
+        '#value' => $event_id,
+      ];
+
+      $element['ids']['event_maker_id'] = [
+        '#type' => 'value',
+        '#value' => $event_maker_id,
+      ];
+
+      $element['ids']['price_id'] = [
+        '#type' => 'value',
+        '#value' => $price_id,
+      ];
+      break;
+  }
+
+  return $element;
+}
+
+/**
+ * Implements hook_field_presave().
+ */
+function ding_place2book_field_presave($entity_type, $entity, $field, $instance, $langcode, &$items) {
+  switch ($field['type']) {
+    case 'ding_p2b':
+      if (!empty($items)) {
+        $p2b = P2b\P2b::getInstance();
+        $settings = reset($items);
+        list($event_id, $event_maker_id, $price_id) = array_values($settings['ids']);
+        if (empty($event_id)) {
+          $event_maker_id = _ding_place2book_get_event_maker($entity);
+          $date = field_get_items('node', $entity, 'field_ding_event_date');
+          $start = strtotime($date[0]['value']);
+          if ($event_maker_id && round(abs(time() - $start) / 60, 2) != 0) {
+            try {
```

Comment:
This is a really big block wrapped in `try`.
Recommend to wrap only code that actually might throw an exception.